### PR TITLE
[SYCL][CUDA][TEST-E2E] enable sycl-ls-gpu-default-any on CUDA

### PR DESCRIPTION
**Pull request by JackAKirk**
This is a migration of this PR https://github.com/intel/llvm-test-suite/pull/1144/commits

### CODE REVIEW

File: sycl/test-e2e/Plugin/sycl-ls-gpu-default-any.cpp [hunk 1]
```diff
@@ -21,6 +21,6 @@
 // is present.
 // The test crashed on CUDA CI machines with the latest OpenCL GPU RT
 // (21.19.19792).
```
Comment (al42and) [inline sycl/test-e2e/Plugin/sycl-ls-gpu-default-any.cpp:23]: Shall this comment be updated?

Comment (JackAKirk) [reply sycl/test-e2e/Plugin/sycl-ls-gpu-default-any.cpp]: Thanks for pointing this out. I've removed this out of date comment.

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Comment by JackAKirk** [comment]
This should be merged.

**Pull request merged**